viscy/representation/lca.py

```diff
@@ -68,27 +71,50 @@ def __init__(
             Train/validate/test split ratio, must sum to 1.
         batch_size : int
             Batch sizes
+        use_smote : bool, optional
+            Whether to apply SMOTE to the training data, by default False
         """
         super().__init__()
         if not embeddings.shape[0] == labels.shape[0]:
             raise ValueError("Number of samples in embeddings and labels must match.")
         if sum(split_ratio) != 1.0:
             raise ValueError("Split ratio must sum to 1.")
-        self.dataset = TensorDataset(embeddings.float(), labels.long())
+        indices = torch.arange(embeddings.shape[0])
+        self.dataset = TensorDataset(embeddings.float(), labels.long(), indices)
```

Comment:
This is not used? `torch.utils.data.Subset` has `indices` as an attribute.

Reply:
Hm ok, I need to know the indicies of the test set from the original dataset such that I can know which cells to populate predictions for in the output csv. Is there another approach to do this?

I am using indicies in train_and_test_linear_classifier (line 253)

test_indices = data.test_indices

Reply:
That is exactly what I'm referring to. You are using the attribute of `Subset`, which is not the value of the dataset. The third tensor value is never used.

Reply:
One idea is to do the class balancing and splitting outside in numpy/sklearn. That way we can re-use this for the sklearn models (logistic regression, KNN etc) too.

Reply:
Do you want to refactor to include this?

Reply:
I was going to do this but then you'd have to change LCA.py more as the splitting would be done manually outside in the script. If you could refactor to include this and make sure functionality of being able to save a csv stays the same, that would be good.

Reply:
Do you still want to fix this PR and merge it before refactoring?

Reply:
Yes, that would be ideal so we can start making figures with it. What is the minimal change here for the indicies you'd like me to do?

Reply:
Use the `Subset.indices` attribute directly and avoid adding the indices as a value of the dataset.